rs/replicated_state/src/canister_state/system_state/call_context_manager.rs

```diff
@@ -423,6 +423,14 @@ pub struct CallContextManager {
     /// paused or aborted DTS response execution, if any.
     callbacks: BTreeMap<CallbackId, Arc<Callback>>,
 
+    /// Callback deadline priority queue. Holds all not-yet-expired best-effort
+    /// callbacks, ordered by deadline. `CallbackIds` break ties, ensuring
+    /// deterministic ordering.
+    ///
+    /// When a `CallbackId` is returned by `expired_callbacks()`, it is removed from
+    /// the queue. This ensures that each callback is expired at most once.
+    unexpired_callbacks: BTreeSet<(CoarseTime, CallbackId)>,
```

Comment:
Would it make sense (or be possible without a huge refactoring) to encapsulate `callbacks` together with `unexpired_callbacks` in a dedicated struct to make the invariants even better encapsulated?

Reply:
As things stand, the invariant we have is that `unexpired_callbacks` is a subset of the best-effort callbacks in `callbacks`. And there are exactly two places where we mutate the two collections: `register_Callback()` and `unregister_callback()`.

So we could simply put the two into a `struct` (even place it in its own sub-module, so we cannot access the two fields directly from `CallContextManager`) but I'm not convinced it would buy us much.

It's your call.

Reply:
No strong opinion either. Was mainly wondering. Lets leave it as you have written it now.